torch/_refs/__init__.py

```diff
@@ -3377,6 +3378,49 @@ def empty_strided(
     )
 
 
+@register_decomposition(torch.ops.aten.eye)
+@out_wrapper()
+def eye(
+    n: int,
+    m: Optional[int] = None,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    layout: torch.layout = torch.strided,  # TODO: unused, see below
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,  # TODO: unused, see below
+    requires_grad: bool = False,
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.eye
+    """
+    if m is None:
+        m = n
+
+    check(n >= 0, lambda: f"n must be greater or equal to 0, got {n}")
+    check(m >= 0, lambda: f"m must be greater or equal to 0, got {m}")
+
+    range_n = torch.arange(
+        0, n, dtype=torch.float64, device=device, requires_grad=requires_grad
+    )
+    range_m = torch.arange(
+        0, m, dtype=torch.float64, device=device, requires_grad=requires_grad
+    )
+
+    cond = range_n.unsqueeze(-1) == range_m
```

Comment:
Tensors `range_n` and `range_m` do not require grads, they are for indexing purposes only. Moreover, it is not safe to compare floating point numbers via `==` operator.
Please use `requires_grad=False` and `dtype=torch.int64` for both of these tensors.